trusted.ci.jenkins.io.tf

```diff
@@ -122,6 +124,72 @@ resource "azurerm_linux_virtual_machine" "trusted_bounce" {
   }
 }
 
+# CONTROLLER VM
+## NETWORK INTERFACE with internal ip
+resource "azurerm_network_interface" "trusted_controller" {
+  name                = "trusted-controller"
+  location            = azurerm_resource_group.trusted_ci_jenkins_io_controller.location
+  resource_group_name = azurerm_resource_group.trusted_ci_jenkins_io_controller.name
+  tags                = local.default_tags
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = data.azurerm_subnet.trusted_controller.id
+    private_ip_address_allocation = "Dynamic"
```

Comment:
I wonder if we should not select "static" and define a static IP in the subnet, as it is the case today: https://github.com/jenkins-infra/runbooks/tree/main/trusted-ci#connecting 
(otherwise the ssh config of all users would have to change on each VM restart where the private IP could change).
Unless I misunderstood the lifecycle of this ip dynamic allocation (maybe it is only if we chaneg the NIC)?

A safer road would be to define an (internal) DNS record in this subnet (e.g. only inside this private vnet) pointing to the private ip of the VM: in that case no problem to have it dynamically defined, as soon as we let the DNS record to always match the IP or the VM.

WDYT?

(Non blocking comment: we can start with the actual code)